Proteomics/ProteolyticDigestion/CompactPeptideBase.cs

```diff
@@ -190,16 +190,23 @@ protected static IEnumerable<double> ComputeFollowingFragmentMasses(PeptideWithS
                 // If modification exists
                 if (yyy.AllModsOneIsNterminus.TryGetValue(oneBasedIndexToLookAt + 1, out currentModification))
                 {
-                    if (currentModification.neutralLosses.Count == 1 && oneBasedIndexToLookAt != 0 && oneBasedIndexToLookAt != yyy.Length + 1)
+                    List<double> neutralLossValuesList = new List<double>();
```

Comment:
I don't really understand the changes here, can you explain this?

Creating these lists for each compact peptide (actually I think it's making new lists for each fragment in the peptide...) may cause serious RAM problems

Reply:
Here is a place where we need to have access to or pass the dissociation type. Then we can access the values in the dictionary key. This code is wrong and needs to be fixed.